generator/feature_segments_checker/feature_segments_checker.cpp

```diff
@@ -212,20 +239,37 @@ class Processor
     if (realFeatureLengthMeters == 0.0)
       return;
 
-    double distFromStartMeters = 0;
+    double const k = (endAltitude - startAltitude) / realFeatureLengthMeters;
     for (uint32_t i = 1; i + 1 < numPoints; ++i)
     {
-      // Feature segment length.
-      double const segmentLengthMeters =
-          MercatorBounds::DistanceOnEarth(f.GetPoint(i - 1), f.GetPoint(i));
-      distFromStartMeters += segmentLengthMeters;
-
-      generator::SrtmTile::THeight const pointAltitude =
-          m_srtmManager.GetHeight(MercatorBounds::ToLatLon(f.GetPoint(i)));
-      int32_t const deviation = static_cast<int32_t>(ExpectedPointAltitude(startAltitude, endAltitude, distFromStartMeters,
-                                                                           realFeatureLengthMeters)) - pointAltitude;
+      int32_t const deviation =
+          static_cast<int32_t>(ExpectedPointAltitude(k, startAltitude, pointDists[i])) - pointAltitudes[i];
       m_diffFromLinear[deviation]++;
     }
+
+    // Linear least squares for feature points.
+    {
+      long double const distSum = accumulate(pointDists.begin(), pointDists.end(), 0.0);
```

Comment:
A lot of misleading names. Please, create an individual method that takes a vector of xs and vector of ys and returns coeffs for linear model.